db/sql/00_msar.sql

```diff
@@ -2392,6 +2458,73 @@ END;
 $$ LANGUAGE plpgsql;
 
 
+CREATE OR REPLACE FUNCTION
+msar.get_preview(
+  tab_id oid,
+  col_cast_def jsonb,
+  rec_limit integer
+) RETURNS jsonb AS $$/*
+Preview a table, applying different type casts and options to the underlying columns before import,
+returning a JSON object describing the records of the table.
+
+Note that these casts are temporary and do not alter the data in the underlying table,
+if you wish to alter these settings permanantly for the columns see msar.alter_columns.
+
+Args:
+  tab_id: The OID of the table to preview.
+  col_cast_def: A JSON object describing the column settings to apply.
+  rec_limit (optional): The upper limit for the number of records to return.
+
+The col_cast_def JSONB should have the form:
+[
+  {
+    "attnum": <int>,
+    "type": {
+      "name": <str>,
+      "options": {
+        "length": <integer>,
+        "precision": <integer>,
+        "scale": <integer>
+        "fields": <str>,
+        "array": <boolean>
+      }
+    },
+  },
+  {
+    ...
+  },
+  ...
+]
+*/
+DECLARE
+  tab_name text;
+  sel_query text;
+  records jsonb;
+BEGIN
+  tab_name := __msar.get_relation_name(tab_id);
+  sel_query := 'SELECT id, %s FROM %s LIMIT %L';
+  WITH preview_cte AS (
+    SELECT string_agg(
+      'CAST(' ||
+      __msar.build_cast_expr(
+        msar.get_column_name(tab_id, (col_cast ->> 'attnum')::integer), col_cast -> 'type' ->> 'name'
+      ) ||
+      ' AS ' ||
+      msar.build_type_text(col_cast -> 'type') ||
+      ')'|| ' AS ' || msar.get_column_name(tab_id, (col_cast ->> 'attnum')::integer),
+      ', '
+    ) AS cast_expr
+    FROM jsonb_array_elements(col_cast_def) AS col_cast
+    WHERE NOT msar.is_mathesar_id_column(tab_id, (col_cast ->> 'attnum')::integer)
```

Comment:
There's a bug here (or at least it throws an exception when the old function didn't) whenever the table doesn't have an `id` column. I think you should extract some kind of `get column or cast expression` function from this so you can sort that out more smoothly (similarly to the python version; it should just return the column name for its expression when appropriate, and the cast blob otherwise).

Reply:
I don't think I quite understand the bug you are referring to, since the function(`get_preview`) is meant to be able to preview a mathesar table which is guaranteed to have an `id` col, I've taken the liberty to assume that it'll always be present for a table which is being previewed. 

> it should just return the column name for its expression when appropriate, and the cast blob otherwise

What's the downside of having a cast blob for a column where both target and column_type are same? I guess it would be a little efficient if we only use column name in such case but is it worth giving up the readability?

Reply:
I suppose I'm okay with just casting all columns (including `id`). I realize that in our current UI flow, the preview is only used for importing, but that's not enforced by the API, and the function itself is quite useful for previewing any proposed change to the column types of a table. Thus, I'd prefer avoiding breaking the current functionality that lets the function be used against any table regardless of whether it has an `id` column.

Reply:
Cool, I've made the requested changes in a20e663.